src/go/rdctl/pkg/config/config.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright © 2022 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package config handles all the config-related parts of rdctl
+
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/spf13/cobra"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strconv"
+)
+
+// The CLIConfig struct is used to store the json data read from the config file.
+type CLIConfig struct {
+	User     string
+	Password string
+	Port     int
+}
+
+var (
+	user     string
+	host     string
+	port     string
+	password string
+
+	configDir           string
+	configPath          string
+	defaultConfigPath   string
+	deferredConfigError error
+)
+
+// DefineGlobalFlags sets up the global flags, available for all sub-commands
```

Comment:
The comment should be changed, I think, to say something about configs.  `root.go` is free to add even more flags.

Maybe rename the function too?

Reply:
I don't understand this comment. Of course more global flags can be defined in `root.go`, but the name of the function strongly suggests it shouldn't.

Right now all the global flags are for connection parameters, but that could be expanded in the future. I think the names in this package are general enough to allow for that.